lib/audit/catalog_to_moab.rb

```diff
@@ -7,14 +7,19 @@ class CatalogToMoab
   # allows for sharding/parallelization by storage_dir
   # use model scope query (which contains ordering), limit for batching, and .each within a while loop to
   # process records in order in batches.  Note that .find_each does batches, but disregards order from
-  # the scope, so we must use .each
+  # the scope, so we must use .each on a limit to process a batch after ordering over the whole set.
   def self.check_version_on_dir(last_checked_b4_date, storage_dir, limit=Settings.c2m_sql_limit)
     start_msg = "#{Time.now.utc.iso8601} C2M check_version starting for #{storage_dir}"
     puts start_msg
     Rails.logger.info start_msg
-    num_to_process = PreservedCopy.least_recent_version_audit(last_checked_b4_date, storage_dir).count
+
+    # all_processable_copies is an AR relation; fine to run it with a .count or a .limit tacked on, but
+    # don't call .each directly on it and get the whole result set at once.
+    all_processable_copies =
+      PreservedCopy.least_recent_version_audit(last_checked_b4_date).by_storage_location(storage_dir)
```

Comment:
did some testing to satisfy my own curiosity, and the chained scope version does generate essentially the same sql query as the version where `least_recent_version_audit` to `storage_dir`:
```sql
-- [5] pry(main)> PreservedCopy.least_recent_version_audit(Time.now.utc).by_storage_location('spec/fixtures/storage_root01/moab_storage_trunk')
SELECT "preserved_copies".*
FROM "preserved_copies"
INNER JOIN "endpoints"
ON "endpoints"."id" = "preserved_copies"."endpoint_id"
WHERE (last_version_audit IS NULL or last_version_audit < '2018-02-24 05:42:05.055725')
AND "endpoints"."storage_location" = $1
ORDER BY last_version_audit IS NOT NULL, last_version_audit ASC  [["storage_location", "spec/fixtures/storage_root01/moab_storage_trunk"]]

-- [1] pry(main)> PreservedCopy.least_recent_version_audit(Time.now.utc, 'spec/fixtures/storage_root01/moab_storage_trunk')
SELECT "preserved_copies".*
FROM "preserved_copies"
INNER JOIN "endpoints"
ON "endpoints"."id" = "preserved_copies"."endpoint_id"
WHERE "endpoints"."storage_location" = $1
AND (last_version_audit IS NULL or last_version_audit < '2018-02-24 05:42:39.866743')
ORDER BY last_version_audit IS NOT NULL, last_version_audit ASC  [["storage_location", "spec/fixtures/storage_root01/moab_storage_trunk"]]
```

(sql formatting edited for readability)